feat: local sql migrations

davemooreuws opened the pull request:
Ability to build and run local migrations on nitric start and run via the "Run migrations" button. 

Closes https://github.com/nitrictech/nitric/issues/660

davemooreuws committed:
feat: local sql migrations

davemooreuws committed:
fix issues with lint

davemooreuws committed:
fix tests

tjholm commented on pkg/cloud/sql/sql.go:
```diff
@@ -236,16 +282,135 @@ func (l *LocalSqlServer) Query(ctx context.Context, connectionString string, que
+		if err != nil {
+			exit.GetExitService().Exit(fmt.Errorf("failed to build and run migrations: %w", err))
+		}
+	}
```
Including automated migrations in an update handle for local resource state might be a touch aggressive? If I'm reading this right, this will trigger migrations for existing and new databases on any resource state change (e.g. Adding a Topic).

For now I think providing a convenient way to manually trigger migrations should suffice, as you've already included in the dashboard.

davemooreuws replied:
Updated

davemooreuws committed:
run migrations only via button

davemooreuws committed:
fix rebase conflict

davemooreuws committed:
fix test

tjholm committed:
WIP: package rework.

add system logs service for logging to the CLI.

remove exit service

add panic recovery for stopping local cloud.

Add missing initial status.

Move connection string changes into migrate method.

tjholm committed:
remove old code.

tjholm committed:
New -> NewService

tjholm committed:
fmt

tjholm commented:
@davemooreuws have made changes to move service/migrations back to the project package for now. I've also changed to exit service to a system logging service that logs in the CLI along with running nitric services as the "nitric" service in order to side log runtime errors rather than exiting the CLI for now.

Can you re-test to make sure everything is working as you expect.

davemooreuws commented:
> @davemooreuws have made changes to move service/migrations back to the project package for now. I've also changed to exit service to a system logging service that logs in the CLI along with running nitric services as the "nitric" service in order to side log runtime errors rather than exiting the CLI for now.
> 
> Can you re-test to make sure everything is working as you expect.

Tested, works great!
